### DISCUSSION TITLE
Environment variable reference

### DISCUSSION THREAD

**Discussion by dawoon-coffeechat** [vercel/community#5035] [Help] [answered]
### Summary

At some point, when you do a production build, the development environment variables are applied and built. I haven't touched any build-related settings, and the same phenomenon occurs with newly added projects. Are there any changes to vercel's settings? Because of this problem, I am experiencing the inconvenience of having to roll back every time I build. Please check.

### Steps to Reproduce

production build

**Comment by amyegan** [comment, marked as answer]
Does your project use Turborepo or Nx? It's possible that this is a remote cache problem. The Turborepo docs have a [Configuring Cache Outputs](https://turbo.build/repo/docs/core-concepts/caching/what-to-cache#configuring-cache-outputs) page and the Nx docs have a [How Caching Works](https://nx.dev/concepts/how-caching-works#runtime-hash-inputs) page. Those pages should help if caching is the source of the problem.

If that's not it, I could use a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) to dig into the problem further.

**Reply by dawoon-coffeechat** [reply]
thank you very much
I was using turborepo and the cache-related settings were set as you told me. Thanks to you, I solved it. 

Thank you amyegan!
